src/WebJobs.Script/Description/Script/ScriptFunctionInvoker.cs

```diff
@@ -77,59 +76,43 @@ public static bool IsSupportedScriptType(ScriptType scriptType)
+            string functionInstanceOutputPath = Path.Combine(Path.GetTempPath(), "Functions", "Binding", invocationId);
 
-                object convertedInput = ConvertInput(input);
-                Utility.ApplyBindingData(convertedInput, binder.BindingData);
-                Dictionary<string, object> bindingData = binder.BindingData;
-                bindingData["InvocationId"] = invocationId;
+            Dictionary<string, string> environmentVariables = new Dictionary<string, string>();
+            InitializeEnvironmentVariables(environmentVariables, functionInstanceOutputPath, input, _outputBindings, functionExecutionContext);
 
-                await ProcessInputBindingsAsync(convertedInput, functionInstanceOutputPath, binder, _inputBindings, _outputBindings, bindingData, environmentVariables);
+            object convertedInput = ConvertInput(input);
+            Utility.ApplyBindingData(convertedInput, binder.BindingData);
+            Dictionary<string, object> bindingData = binder.BindingData;
+            bindingData["InvocationId"] = invocationId;
 
-                // TODO
-                // - put a timeout on how long we wait?
-                // - need to periodically flush the standard out to the TraceWriter
-                Process process = CreateProcess(path, workingDirectory, arguments, environmentVariables);
-                process.Start();
-                process.WaitForExit();
+            await ProcessInputBindingsAsync(convertedInput, functionInstanceOutputPath, binder, _inputBindings, _outputBindings, bindingData, environmentVariables);
 
-                string output = process.StandardOutput.ReadToEnd();
-                TraceWriter.Info(output);
-                traceWriter.Info(output);
-
-                startedEvent.Success = process.ExitCode == 0;
-
-                if (!startedEvent.Success)
+            Process process = CreateProcess(path, workingDirectory, arguments, environmentVariables);
+            process.OutputDataReceived += (s, e) =>
+            {
+                if (e.Data != null)
                 {
-                    string error = process.StandardError.ReadToEnd();
-                    throw new ApplicationException(error);
+                    TraceWriter.Info(e.Data);
+                    traceWriter.Info(e.Data);
                 }
+            };
 
-                await ProcessOutputBindingsAsync(functionInstanceOutputPath, _outputBindings, input, binder, bindingData);
+            process.Start();
+            process.BeginOutputReadLine();
 
-                TraceWriter.Info(string.Format("Function completed (Success, Id={0})", invocationId));
-            }
-            catch
-            {
-                startedEvent.Success = false;
-                TraceWriter.Error(string.Format("Function completed (Failure, Id={0})", invocationId));
-                throw;
-            }
-            finally
+            await Task.Run(() => process.WaitForExit());
```

Comment:
We should consider not using `Task.Run` here and using the `Exited` event to asynchronously wait for the process to finish. An example of that may be found in the PackageManager here:
https://github.com/fabiocav/azure-webjobs-sdk-script/blob/546f79bdb40da4fdf8b3d34f14f2938f18f00172/src/WebJobs.Script/Description/DotNet/PackageManager.cs#L56-L76